diff --git a/components/javascript_dialogs/tab_modal_dialog_manager.cc b/components/javascript_dialogs/tab_modal_dialog_manager.cc
index 7c6cf376decfd31b108d1b93081c9c2b910edcdf..cdfeca39d3a0300ac6afffad54f990a369b38fa5 100644
--- a/components/javascript_dialogs/tab_modal_dialog_manager.cc
+++ b/components/javascript_dialogs/tab_modal_dialog_manager.cc
@@ -217,6 +217,8 @@ void TabModalDialogManager::RunBeforeUnloadDialog(
   DCHECK_EQ(web_contents,
             content::WebContents::FromRenderFrameHost(render_frame_host));
 
+  BRAVE_TAB_MODAL_DIALOG_MANAGER_RUN_BEFORE_UNLOAD_DIALOG
+
   // onbeforeunload dialogs are always handled with an app-modal dialog, because
   // - they are critical to the user not losing data
   // - they can be requested for tabs that are not foremost
@@ -253,6 +255,7 @@ void TabModalDialogManager::CancelDialogs(content::WebContents* web_contents,
 
 void TabModalDialogManager::OnVisibilityChanged(
     content::Visibility visibility) {
+  BRAVE_TAB_MODAL_DIALOG_MANAGER_ON_VISIBILITY_CHANGED
   if (visibility == content::Visibility::HIDDEN) {
     HandleTabSwitchAway(DismissalCause::kTabHidden);
   } else if (pending_dialog_) {
